pennylane/transforms/metric_tensor.py

```diff
@@ -310,33 +358,16 @@ def wrapper(*args, **kwargs):
 
         kwargs.pop("shots", False)
         cjac = cjac_fn(*args, **kwargs)
+        if qnode.interface == "autograd":
+            trainable_args = np.where([qml.math.requires_grad(arg) for arg in args])[0]
+            shape = qml.math.shape(args[trainable_args[0]])
+            cjac_hotfix = len(trainable_args) > 1 and all(
+                qml.math.shape(args[i]) == shape for i in trainable_args[1:]
+            )
+        else:
+            cjac_hotfix = False

```

Comment:
@dwierichs could you add a descriptive `# TODO:` comment here detailing the issue, what the hotfix is, and how it can be addressed/removed?

This will help future developers that are looking at the code, and wondering what `cjac_hotfix` refers to 😆

Reply:
@dime10 you might also find this useful!

Reply:
Yes, good idea, will do!

Reply:
Thanks, reverting the stack&transpose is a nice way to solve the problem! I took a slightly different approach for the [gradient transforms](https://github.com/PennyLaneAI/pennylane/blob/1389303570e1e11079cf68f755412b00b4095674/pennylane/gradients/gradient_transform.py#L170-L196) in order to match the derivatives returned by `qml.jacobian`.

Reply:
I see. Maybe I should do the same. I'm planning to implement the change in `qml.jacobian` very soon and then we can let go of all these hacks :sweat_smile: 
I'll go with your approach for now, then.

Reply:
Yes that will be a lot nicer for sure! :D

But I don't know, your approach is significantly nicer as an intermediary solution and if there isn't the expectation for the metric tensor to behave like qml.jacobian, I think that may be the way to go. (I think for the param_shit or param_shift_hessian people would expect to get the same result out as when using qml.jacobian directly.)